app/src/main/java/com/sopt/now/compose/presentation/home/HomeScreen.kt

```diff
@@ -2,35 +2,67 @@ package com.sopt.now.compose.presentation.home
 
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.lazy.LazyColumn
-import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.itemsIndexed
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
 import com.sopt.now.compose.R
 import com.sopt.now.compose.component.text.TextWithRow
 import com.sopt.now.compose.model.Friend
 import com.sopt.now.compose.model.User
 import com.sopt.now.compose.ui.theme.GreenMain
 import com.sopt.now.compose.ui.theme.NOWSOPTAndroidTheme
 import com.sopt.now.compose.ui.theme.YellowMain
+import com.sopt.now.compose.util.UiState
+import com.sopt.now.compose.util.toast
+import timber.log.Timber
 
 @Composable
-fun HomeScreen(user: User) {
+fun HomeScreen(homeViewModel: HomeViewModel = viewModel()) {
+    // 상태 관리를 위한 state 변수 선언
+    val context = LocalContext.current
+    val lifecycleOwner = LocalLifecycleOwner.current
+
+    var userData by remember { mutableStateOf(emptyList<User>()) }
+    var friendData by remember { mutableStateOf(emptyList<Friend>()) }
+
+    LaunchedEffect(homeViewModel.getUserList, lifecycleOwner) {
+        homeViewModel.getUserList.observe(lifecycleOwner) {
+            when (it) {
+                is UiState.Success -> {
+                    userData = homeViewModel.mockUserList
+                    friendData = it.data ?: emptyList()
+                }
+
+                is UiState.Failure -> context.toast("유저 리스트 조회 실패 : ${it.errorMessage}")
+                is UiState.Loading -> Timber.d("로딩중")
+            }
+        }
+    }
+
     LazyColumn(
         modifier = Modifier
             .fillMaxSize()
     ) {
         // User
-        item {
+        itemsIndexed(userData) { index, user ->
             TextWithRow(
                 R.drawable.img_user_profile_dororo,
                 "user profile img",
                 75.dp,
```

Comment:
nit : 아래처럼 각각 어떤 파라미터인지 적어주시면 코드 읽을 때 조금 편할 것 같아요!
```suggestion
            TextWithRow(
                painter = R.drawable.img_user_profile_dororo,
                contentDescription = "user profile img",
                profileSize  = 75.dp,
```

Reply:
가독성 측면에서 훨씬 좋을 것 같네요! 반영했습니다 ㅎㅎ